packages/protocol/src/session/InsecureSession.ts

```diff
@@ -95,6 +90,7 @@ export class InsecureSession extends Session {
     }
 
     async end() {
-        await this.closeCallback?.();
+        logger.info(`End insecure session ${this.name}`);
+        this.manager?.insecureSessions.delete(this.nodeId);
```

Comment:
really such an access from "outside"? session could also emit an event when closed and sessionmanager reacts and deletes? Same on SecureSession do we really want to wire the sessionmanager into the session? That somehow feels wrong

Reply:
Right now this also replaces the "everything in the world" context object required by the CommissioningServer code branch.  Looks like we could add a "closed" promise to the session and have the manager listen to that though I'd maybe wait until CommissioningServer goes away.

Reply:
I did make the manager getter protected so this is just an implementation detail now.